zaza/openstack/charm_tests/mysql/tests.py

```diff
@@ -1145,3 +1146,64 @@ def test_910_restart_on_config_change(self):
             {}, {},
             self.services)
         logging.info("Passed restart on changed test.")
+
+    def test_action_bootstrap_mysqlrouter(self):
+        """Checking bootstrap-mysqlrouter action.
+
+        Run the bootstrap-mysqlrouter action
+        """
+        logging.info("Starting bootstrap-mysqlrouter test")
+
+        # put empty string to conf_file and make it wrong
+        logging.info("Breaking configuration file")
+        zaza.model.run_on_leader(self.application,
+                                 "echo '[DEFAULT]\n \
+                                        [metadata_cache:[\\w$]+$] \
+                                        ' > {}".format(
+                                     self.conf_file))
+
+        # update status to make the status error
+        logging.info("Run update-status")
+        self.run_update_status_hooks(['keystone-mysql-router/0'])
+
+        while True:
+            time.sleep(5)
```

Comment:
This is a pattern that we try to avoid in zaza-openstack-tests.  You can create the same effect using the`tenacity` module.  Take a look at https://github.com/openstack-charmers/zaza-openstack-tests/blob/a07fb4017a2c7a941d64fc6589bfc4ea6c1203b8/zaza/openstack/charm_tests/designate/utils.py#L135 for inspriration.  Essentially, you want something along the lines of an exponential back-off waiting say 2 minutes before bailing out and raising an error to stop the test.  Otherwise, this could hold here for the full 120 minutes (the maximum a test run can last) holding up other tests in the CI system.